src/rust/ide/enso-protocol/src/language_server/types.rs

```diff
@@ -37,6 +37,14 @@ impl Display for Path {
 }
 
 impl Path {
+    /// Splits path into name and segments. e.g.:
+    /// Path{root_id,segments:["foo","bar","qux"]} => ("qux",Path{root_id,segments:["foo","bar"]})
+    pub fn split_into_name_and_segments(&self) -> (String,Path) {
```

Comment:
If we return by value, we should take `self` by value as well. This way we'll avoid clones.